### ISSUE TITLE
Catalogs cannot be loaded via resource: URL

### ISSUE THREAD

**Issue by sewe** [mojohaus/xml-maven-plugin#58]
The FAQ state the [following](https://www.mojohaus.org/xml-maven-plugin/faq.html#ClassLoader_Resources):
>The plugin attempts to resolve resources through the class loader automatically. Alternatively, you can enforce using the class loader, if you specify your resource with an URI like "resource:org/foo/stylesheet.xsl".

Unfortunately, that is only true to stylesheets or schemas, not for catalog files. (To test, simply change `mojo-1438-validate`’s `pom.xml`: `<catalog>resource:src/main/xml/catalog.xml</catalog>`)

This prevents the following use case:

1. Bundle your schemas or stylesheets along with a catalog into a resource JAR
2. Refer to the schemas only via their `<publicId>` during plug-in `<configuration>`
3. Moreover, any schema imports can again only use their `xs:import/@namespace`; no `xs:import/@schemaLocation` with a Maven-specific `resources:` scheme is necessary, as the catalog takes care of the mapping.

In other words, the schemas are completely generic. Any Maven-specifics are localized in the `catalog.xml` that is shared along-side the schemas.

**Comment by jochenw** [comment]
Just a hint: Unless the current directory is in your class path (unlikely for a Maven project), then src/main/xml/catakig.xml is an invalid URI.
